### PR TITLE
[Commands] Propogate error codes from SwiftTool runImpl

### PR DESCRIPTION
**Pull request by bhargavg**

### CODE REVIEW

**Comment by bhargavg** [comment]
@swift-ci please smoke test

File: Sources/Commands/SwiftTestTool.swift [hunk 1]
```diff
@@ -148,16 +149,17 @@ public class SwiftTestTool: SwiftTool<TestToolOptions> {
...
-            exit(ranSuccessfully ? 0 : 1)
+            return ranSuccessfully ? .success : .failure(1)
 
         case .runParallel:
             let testPath = try buildTestsIfNeeded(options)
             let testSuites = try getTestSuites(path: testPath)
             let tests = testSuites.filteredTests(specifier: options.testCaseSpecifier)
             let runner = ParallelTestRunner(testPath: testPath, processSet: processSet)
             try runner.run(tests)
-            exit(runner.ranSuccesfully ? 0 : 1)
```
Comment (aciidb0mb3r) [inline Sources/Commands/SwiftTestTool.swift]: We need to return the failure here, right?

File: Sources/Commands/SwiftTool.swift [hunk 2]
```diff
@@ -267,12 +267,17 @@ public class SwiftTool<Options: ToolOptions> {
...
             // Print any non fatal diagnostics like warnings, notes.
             printDiagnostics()
+
+            switch status {
+            case .success: POSIX.exit(0)
+            case let .failure(code): POSIX.exit(Int32(code))
+            }
         } catch {
             printDiagnostics()
             handle(error: error)
```
Comment (aciidb0mb3r) [inline Sources/Commands/SwiftTool.swift]: We should remove the `exit(1)` from this method and move the above switch statement after the `try..catch`.

File: Sources/Commands/SwiftTool.swift [hunk 3]
```diff
@@ -419,6 +424,11 @@ public class SwiftTool<Options: ToolOptions> {
             )
         })
     }()
+
+    enum ExecutionStatus {
```
Comment (aciidb0mb3r) [inline Sources/Commands/SwiftTool.swift]: doc comment plz

File: Sources/Commands/SwiftTool.swift [hunk 4]
```diff
@@ -419,6 +424,11 @@ public class SwiftTool<Options: ToolOptions> {
...
+        case success
+        case failure(Int)
```
Comment (aciidb0mb3r) [inline Sources/Commands/SwiftTool.swift]: We only have one exit code currently, right? If so, lets remove the (Int) for now.

File: Sources/Commands/SwiftTool.swift [hunk 5]
```diff
@@ -262,12 +263,13 @@ public class SwiftTool<Options: ToolOptions> {
 
     /// Execute the tool.
     public func run() {
+        var status = SwiftTool.ExecutionStatus.failure
```
Comment (aciidb0mb3r) [inline Sources/Commands/SwiftTool.swift]: We can make this `let` and assign values in both do and catch.

File: Sources/Commands/SwiftTool.swift [hunk 6]
```diff
@@ -419,6 +426,12 @@ public class SwiftTool<Options: ToolOptions> {
             )
         })
     }()
+
+    /// An enum indicating the execution status of run commands
```
Comment (aciidb0mb3r) [inline Sources/Commands/SwiftTool.swift]: nit, end the commit with punctuation.

File: Sources/Commands/SwiftTool.swift (hunk 5 above)
Comment (bhargavg) [inline Sources/Commands/SwiftTool.swift]: I've tried it before, doesn't seem to work

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Pull request merged**